plugins/module_utils/botocore.py

```diff
@@ -54,6 +54,56 @@
 from ansible.module_utils.six import text_type
 
 from .retries import AWSRetry
+from .version import LooseVersion
+
+
+class BotocoreBaseMixin(object):
+    _botocore_versions = None
+
+    @property
+    def minimum_supported_boto3(self):
+        return '1.17.0'
```

Comment:
What about pulling botocore and boto3 versions directly from requirements.txt?

Reply:
I like the idea, I don't know if we can guarantee it'll be available on the "remote" side...

Reply:
@jillr Do you happen to know it would be possible to pull the supported boto3/botocore versions from requirements.txt?  

I'm assuming that because modules run "remote" side the file won't be there to read...

Reply:
When Ansible runs a module on a remote host, it first loads the module and try to figure out what are the Python dependencies. From that, it includes all the dependencies in the AnsibleZ zip file. If we want to include a meta information like this, I'm afraid we need it to be present in a Python file.

We can script the generation of a plugins/module_utils/versions.py and run it during our release process. But this may sounds like an extra source of burden.